### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by kinsaurralde**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -13,8 +13,17 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:16]: You can leave this empty line to separate "package" and "import" blocks.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: datastore can be a private member

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -23,9 +32,48 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: datastore can be a private member

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -23,9 +32,48 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: do you need to update this comment?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -23,9 +32,48 @@
...
+      datastore.put(commentEntity);
+    }
+    response.sendRedirect("/#comments");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: maybe rename this function to a meaningful name, e.g. getRequestParameterOrDefault

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Google LLC
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: 2020

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      datastore.delete(entity.getKey());
+    }
+    response.sendRedirect("/#comments");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: Is this function called somewhere? It's better to be added when it actually gets called, so we will decide how to design this function based on how it's used.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -23,9 +32,48 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Kyle!</h1>");
+    int numComments = Integer.parseInt(getParameter(request, "num-comments", "5"));
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The defaultValue can be a private static member variable in this class. I noticed that you have different default values, so you can create different variables such as DEFAULT_COMMENTS_COUNT

You can check the style guide https://google.github.io/styleguide/javaguide.html to use the correct style.

**Review by huzcn — CHANGES_REQUESTED** [review]
Added a few comments particularly in java code.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 7 above)
Comment (kinsaurralde) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: deleted now

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 9]
```diff
@@ -29,16 +30,19 @@
...
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that adds and returns comments */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static String DEFAULT_COMMENTS_COUNT = "5";
+  private static String DEFAULT_PAGE = "1";
+  private static String DEFAULT_COMMENT = "";
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: constants can use "private static final String" modifier.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 10]
```diff
@@ -29,16 +30,19 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that adds and returns comments */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: this can be "final" since the object never changes.

**Review by huzcn — APPROVED** [review]
Looks nice. Approve for merging after nits are resolved.

**Pull request merged**